### PR TITLE
Javascript tests

### PR DESCRIPTION
**Pull request by AlexandraImbrisca**
Creates the environment to set up the JavaScript tests using [Jasmine](https://jasmine.github.io/). I encountered multiple issues while trying to incorporate Jasmine through Maven. After all, I decided to create **a script that will install Jasmine and run the tests at every build**. The script will install Jasmine (if it is not already installed), create the structure of the Jasmine project and run the tests. 

The tests included are focused on the **utilities functions used for generating the calendar.**

_There is still a problem related to the .mjs extension. The file is not automatically seen as a JavaScript file if it has this extension (the server sets the content-type as text/plain instead of application/javascript). We consider looking further into that later, but at the moment we manually created the module file._

**Note: It also includes helper functions to get/set a cookie in case we will later incorporate cookie related tests (setting/getting the id_token cookie).**

### CODE REVIEW

File: src/test/javascript/CalendarUtilities.mjs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+  if (currentMonth === 11) {
+    return new Date(currentYear + 1, 0);
+  }
+
+  return new Date(currentYear, currentMonth + 1);
+}
+
+/**
+ * TODO: Include unit tests.
```
Comment (tatiana-s) [inline src/test/javascript/CalendarUtilities.mjs:68]: Aren't there unit tests for this, so the TODO should be removed?

File: src/main/webapp/script.js [hunk 2]
```diff
@@ -37,14 +35,33 @@ function createElement(elementType, className, innerText) {
...
+/**
+ * Gets the value of the cookie with the given name if it is set. Otherwise,
+ * it will return null.
+ * @param {String} cookieName The cookie name.
+ * @return {String} The value associated with the cookie if it was set or null
+ * otherwise.
+ */
+function getCookie(cookieName) {
+  const cookies = document.cookies.split(';');
+
```
Comment (tatiana-s) [inline src/main/webapp/script.js:47]: Just for clarification, is this related to the testing or not?

**Review by tatiana-s — APPROVED** [review]

File: src/main/webapp/script.js (hunk 2 above)
Comment (AlexandraImbrisca) [reply src/main/webapp/script.js:47]: Good point! I included it in case we will later incorporate cookie related tests (setting/getting the id_token cookie).

Comment (csperl) [reply src/main/webapp/script.js:47]: If this is code which is not used at the moment, then I suggest to not add it.
We should not have dead code. (Maintaining code costs money. Having dead code requires you to maintain more code).

File: src/test/javascript/CalendarUtilities.mjs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+class Event {
+  /**
+   * Creates a new event with the given parameters.
+   * @param {Long} id The id used to store the event in the database.
+   * @param {String} title The title of the event.
+   * @param {Date} start The start date of the event.
+   * @param {Date} end The end date of the event.
+   */
+  constructor(id, title, start, end) {
+    /** @private @const {LOng} */
```
Comment (csperl) [inline src/test/javascript/CalendarUtilities.mjs:27]: LOng -> Long

File: src/test/javascript/CalendarUtilities.mjs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+/**
+ * Compares two events based on their start date.
+ * @param {Object} firstEvent First event to be compared.
+ * @param {Object} secondEvent Second event to be compared.
+ * @return {Boolean} False if the two events should be switched.
+ */
+function compareEventsByStartDate(firstEvent, secondEvent) {
+  const firstEventStartDate = firstEvent.start_;
```
Comment (csperl) [inline src/test/javascript/CalendarUtilities.mjs:45]: Nit: Inline the two variables.

**Review by csperl — APPROVED** [review]
